code/game/machinery/fax_machine.dm

```diff
@@ -1,9 +1,319 @@
-var/list/obj/structure/machinery/faxmachine/allfaxes = list()
-var/list/alldepartments = list()
+#define UPLINK_WY "Weyland-Yutani Quantum Relay (tm)"
+#define UPLINK_USCM "USCM High Command Quantum Relay"
+
+#define NETWORK_HC "USCM High Command"
+#define NETWORK_PROVOST "USCM Provost Office"
+#define NETWORK_WY "Weyland-Yutani Corporate Affairs"
+
+#define USCM_NETWORK "USCM Encrypted Network"
+#define COLONY_NETWORK "Colonial Network"
+
+GLOBAL_DATUM_INIT(communications_network, /datum/communications_network, new)
+
+/datum/communications_network
+	/// All uplinks available, uplink name -> uplink
+	var/list/uplinks
+	/// All faxes, network name -> fax
+	var/list/local_networks
+
+/datum/communications_network/New()
+	for(var/datum/uplink/linking as anything in subtypesof(/datum/uplink))
```

Comment:
a better type name than `/datum/uplink` might be nice, e.g. `/datum/fax_faction` or smth